utils/server_auth_test.go

```diff
@@ -0,0 +1,290 @@
+package utils
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+// MockAuthChecker is the mock implementation of AuthChecker.
+type MockAuthChecker struct {
+	FederationCheckFunc func(ctx *gin.Context, token string, scopes []string) error
+	DirectorCheckFunc   func(ctx *gin.Context, token string, scopes []string) error
+	IssuerCheckFunc     func(ctx *gin.Context, token string, scopes []string) error
+}
+
+func (m *MockAuthChecker) FederationCheck(ctx *gin.Context, token string, scopes []string) error {
+	return m.FederationCheckFunc(ctx, token, scopes)
+}
+
+func (m *MockAuthChecker) DirectorCheck(ctx *gin.Context, token string, scopes []string) error {
+	return m.DirectorCheckFunc(ctx, token, scopes)
+}
+
+func (m *MockAuthChecker) IssuerCheck(ctx *gin.Context, token string, scopes []string) error {
+	return m.IssuerCheckFunc(ctx, token, scopes)
+}
+
+// Helper function to create a gin context with different token sources
+func createContextWithToken(cookieToken, headerToken, queryToken string) *gin.Context {
+	r := httptest.NewRequest(http.MethodGet, "/", nil)
+
+	if cookieToken != "" {
+		r.AddCookie(&http.Cookie{Name: "login", Value: cookieToken})
+	}
+	if headerToken != "" {
+		r.Header.Add("Authorization", "Bearer "+headerToken)
+	}
+	if queryToken != "" {
+		q := r.URL.Query()
+		q.Add("authz", queryToken)
+		r.URL.RawQuery = q.Encode()
+	}
+
+	w := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(w)
+	ctx.Request = r
+
+	return ctx
+}
+
+func TestCheckAnyAuth(t *testing.T) {
```

Comment:
Can you add a few more comments to the tests that explain what each test is checking? I'm not grokking what's going on here (most likely my fault, sorry!)